### ISSUE TITLE
[bug]: calling useSpring causes "Uncaught Invariant Violation: Invalid hook call. Hooks can only be called inside of the body of a function component" error

### ISSUE THREAD

**Issue by przemekbroda** [pmndrs/react-spring#1870]
### Which react-spring target are you using?

- [X] `@react-spring/web`
- [ ] `@react-spring/three`
- [ ] `@react-spring/native`
- [ ] `@react-spring/konva`
- [ ] `@react-spring/zdog`

### What version of react-spring are you using?

9.4.4

### What's Wrong?

When I try to use useSpring hook, the app breaks and console shows an error: 
Uncaught Invariant Violation: Invalid hook call. Hooks can only be called inside of the body of a function component.

### To Reproduce

1. Create the app using command npx create-react-app react-spring-test --template typescript
2. Add react-spring using command npm i react-spring
3. Copy and paste a code from https://react-spring.io/basics#basics to App.tsx 
import { useSpring, animated } from 'react-spring'

function App() {
  const props = useSpring({ to: { opacity: 1 }, from: { opacity: 0 } })
  return <animated.div style={props}>I will fade in</animated.div>
}
4. Run the app
5. Open console browser to see the error

### Expected Behaviour

App does not run and browser console shows the error:
2react.development.js:95 Uncaught Invariant Violation: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.
    at invariant (http://localhost:3000/static/js/bundle.js:65843:19)
    at resolveDispatcher (http://localhost:3000/static/js/bundle.js:67228:32)
    at useMemo (http://localhost:3000/static/js/bundle.js:67282:24)
    at useSprings (http://localhost:3000/static/js/bundle.js:40419:61)
    at useSpring (http://localhost:3000/static/js/bundle.js:40512:27)
    at App (http://localhost:3000/static/js/bundle.js:29:72)
    at renderWithHooks (http://localhost:3000/static/js/bundle.js:23944:22)
    at mountIndeterminateComponent (http://localhost:3000/static/js/bundle.js:28520:17)
    at beginWork (http://localhost:3000/static/js/bundle.js:29978:20)
    at HTMLUnknownElement.callCallback (http://localhost:3000/static/js/bundle.js:11904:18)
react-dom.development.js:18525 The above error occurred in the <App> component:

    at App (http://localhost:3000/static/js/bundle.js:29:72)

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
logCapturedError @ react-dom.development.js:18525
react-dom.development.js:26740 Uncaught Invariant Violation: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.
    at invariant (http://localhost:3000/static/js/bundle.js:65843:19)
    at resolveDispatcher (http://localhost:3000/static/js/bundle.js:67228:32)
    at useMemo (http://localhost:3000/static/js/bundle.js:67282:24)
    at useSprings (http://localhost:3000/static/js/bundle.js:40419:61)
    at useSpring (http://localhost:3000/static/js/bundle.js:40512:27)
    at App (http://localhost:3000/static/js/bundle.js:29:72)
    at renderWithHooks (http://localhost:3000/static/js/bundle.js:23944:22)
    at mountIndeterminateComponent (http://localhost:3000/static/js/bundle.js:28520:17)
    at beginWork (http://localhost:3000/static/js/bundle.js:29978:20)
    at beginWork$1 (http://localhost:3000/static/js/bundle.js:34795:18)

### Link to repo

https://github.com/przemekbroda/react-spring-test

**Comment by minaelee** [comment]
Can confirm - I am experiencing the exact same issue since React 18.

**Comment by przemekbroda** [comment]
@minaelee  It looks like react spring version 9.4.5-beta.1 does not have that issue in React 18

**Comment by chinalantern** [comment]
Experiencing the same issue here as well.

**Comment by chinalantern** [comment]
> @minaelee It looks like react spring version 9.4.5-beta.1 does not have that issue in React 18

Confirmed, the beta solves the issue for me.

**Comment by lojithv** [comment]
I got the same issue in a nextjs app after installing a few packages. 
The error came after installing threejs packages. Before that, I used the react-spring konva, animated and web packages without getting any errors. (all of those sub-packages are placed inside the @react-spring package). I think there were build file issues due to package version changes. 

![Screenshot_14](https://user-images.githubusercontent.com/74459314/178112075-8b408c41-7dcc-4c1e-b5c4-ea6aa64dbb9b.png)

In my case, I think it was a cache-related issue. Run the project after removing the ".next" folder fixed the issue. I hope removing the build folder in React will do the same. 

![Screenshot_13](https://user-images.githubusercontent.com/74459314/178112077-e63df56e-bb80-4124-a3a1-1de89a238022.png)

**I hope removing build files before running the app after installing new packages will prevent these kinds of issues.**


And there are two similar functions ("useSpring" and "useSprings"). Make sure to pick the right one for your use case.
